examples/resource-monitoring-app/resource-monitoring-common/src/ReplacementProductListManager.cpp

```diff
@@ -25,7 +26,19 @@
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::ResourceMonitoring;
 
-CHIP_ERROR StaticReplacementProductListManager::Next(Attributes::ReplacementProductStruct::Type & item)
+CHIP_ERROR StaticReplacementProductListManager::Next(Attributes::GenericType & item)
+{
+    if (mIndex < mReplacementProductListSize)
+    {
+        item = mReplacementProductsList[mIndex];
```

Comment:
This actually does not do the right thing, and the only reason it works is because this is returning pointers into infinite-lifetime data.  Please think through where the pointers in `item` end up pointing after this assignment.

The key thing that the "dynamic" case should test is not the "we dynamically built the infinite-lifetime list situation" but the "we read this from flash when asked" situation.  That is, a stack buffer for the string, that then gets filled with the string read from somewhere else, and then we want to hand it out to the caller.